src/layer/water.js

```diff
@@ -112,14 +110,11 @@ export const waterLineIntermittent = {
   id: "water_line_intermittent",
   type: "line",
   minzoom: 8,
-  filter: [
-    "all",
-    ["==", ["get", "intermittent"], 1],
-    ["==", ["get", "class"], "lake"],
-  ],
+  filter: ["all", ["==", ["get", "intermittent"], 1]],
   paint: {
-    "line-color": Color.waterLine,
-    "line-dasharray": [6, 4],
+    "line-color": Color.waterIntermittentOutline,
+    "line-dasharray": [10, 6],
+    "line-width": 0.5,
```

Comment:
Making the line half as thick causes the dash array to shrink by half, so you’ve compensated by making the dashes a little longer. Did you explore making the dashes a little longer, to fully account for the shrinkage? This could also help make the line more legible since the color is lighter than before.

Reply:
Yes I originally set the dash array to [12, 8] to fully compensate for the thickness change.  This works well at integer zoom levels:

[12, 8] at Zoom 15.0:
<img width="313" alt="z15" src="https://user-images.githubusercontent.com/281482/209838117-0b03919b-782c-49dd-aea5-d7a43a9cf705.png">

[12, 8] at Zoom 16.0
<img width="313" alt="z16" src="https://user-images.githubusercontent.com/281482/209838122-62fed5cc-020b-41d6-9695-e3ed9a1e983e.png">

At fractional zoom levels MapLibre stretches the dash array out and it looks significantly different.  The longer dashes and gaps work less well I find.

[12, 8] at Zoom 15.5:
<img width="313" alt="z15 5" src="https://user-images.githubusercontent.com/281482/209838364-b95534a3-6850-4c2b-be31-4b5cf5b3dbf4.png">

[12, 8] at Zoom 15.9:
<img width="313" alt="z15 9" src="https://user-images.githubusercontent.com/281482/209838376-01f51809-4a3e-44a9-a1ce-e5770ebad079.png">


I changed the dash array to [10, 6] in an attempt to make the dashes and gaps at the high end of the fractional zoom levels not quite so long.  Of course, this makes the them a bit too short at the low end.  In the middle they look just right.

[10, 6] at Zoom 15.9:
<img width="313" alt="10,6  at z15 9" src="https://user-images.githubusercontent.com/281482/209838637-27570537-61ba-4a39-9cea-67902197a8bb.png">

[10, 6] at Zoom 15.5:
<img width="313" alt="10,6  at z15 5" src="https://user-images.githubusercontent.com/281482/209840008-b7f3603f-085d-44d0-81a4-5d1f4c987882.png">

[10, 6] at Zoom 15.0:
<img width="313" alt="10,6  at z15" src="https://user-images.githubusercontent.com/281482/209840030-1d9a67db-da3d-40d9-80a5-911cf7974214.png">


Honestly, I'm not very satisfied with this change.  Just due to the way dash arrays are rendered in fractional zoom levels, they will look wrong some of the time.   We can have them look right at the low end and significantly too long at the high end, or right in the middle by too short at the low end and too long a the high end.  Not sure which is least bad.  If you think the former is better, I'm happy to go with that and switch back to [12, 8].

Reply:
> This could also help make the line more legible since the color is lighter than before

My intent was to make the line carry more visual weight, despite being thinner, by darkening and saturating the color.  If you're perceiving it as lighter, perhaps there is more tweaking to do.  It is quite a challenge though, because the line is partially underneath the translucent fill layer.  That lightens whatever line color is chosen somewhat.

Reply:
I hadn’t considered fractional zoom levels. This is yet another thing that mapbox/mapbox-gl-js#12326 fixed… I agree with the compromise you implemented.